### DISCUSSION TITLE
--no-half/vae, FP16/32 and AMD cards

### DISCUSSION THREAD

**Discussion by brknsoul** [vladmandic/sdnext#1041] [Q&A] [answered]
A few commits ago, --no-half,--no-half-vae and FP32 was being forced on AMD cards.
Less than a few commits ago, it wasn't being forced.
Now it is again?

What's the 4-1-1? (Which is USA slang for "what's up with that?")

**Comment by vladmandic** [comment, marked as answer]
`directml` does not support `autocast` so originally you needed to manually set `--no-half`.
then i added auto-detect and forced fp32.
then @lshqqytiger added manual `autocast` routines which looked great, so i removed the forcing of fp32.
then uses noticed that there are still some quality/compatiiblity issues, so i've renabled forcing fp32.
